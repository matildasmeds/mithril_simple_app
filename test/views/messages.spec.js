var m = require("mithril");
var tidy = require("mithril-jest").tidy;
var Messages = require("../../src/views/Messages");
var view = m(Messages);

describe("Messages view", function() {
    afterEach(function() {
	Messages.clear();
    });
    it("Shows success message", function() {
	Messages.show({ success: "Great Success!" });
	var html = tidy(view);
	expect(html).toMatchSnapshot();
    });
    it("Shows info message", function() {
	Messages.show({ info: "There are tulips in Amsterdam."});
	var html = tidy(view);
	expect(html).toMatchSnapshot();
    });
    it("Shows error message", function() {
	Messages.show({ error: "Whoops."});
	var html = tidy(view);
	expect(html).toMatchSnapshot();
    });
    it("Is empty by default", function() {
	var html = tidy(view);
	expect(html).toMatchSnapshot();
    });
});
