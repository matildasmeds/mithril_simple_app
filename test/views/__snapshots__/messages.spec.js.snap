exports[`Messages view Is empty by default 1`] = `
"<div class=\"messages-region\"></div>
"
`;

exports[`Messages view Shows error message 1`] = `
"<div class=\"messages-region\">
  <div class=\"message error-message\">
    Whoops.
  </div>
</div>
"
`;

exports[`Messages view Shows info message 1`] = `
"<div class=\"messages-region\">
  <div class=\"message info-message\">
    There are tulips in Amsterdam.
  </div>
</div>
"
`;

exports[`Messages view Shows success message 1`] = `
"<div class=\"messages-region\">
  <div class=\"message success-message\">
    Great Success!
  </div>
</div>
"
`;
